consentv2.go

```diff
@@ -463,6 +464,16 @@ func parseRange(b *bitReader, validateOnly bool) (map[int]bool, int, error) { //
 				return nil, 0, ErrUnexpectedEnd
 			}
 		}
+		if !validateOnly && !allocated {
+			allocated = true
+			/**
+			We want to get a good idea of how big the vendors map is going to be before adding elements
+			to it, or esle it will grow incrementally and will trigger costly splitting and rehashing
+			events. We use the number of vendros in the first entry as a proxy for the average number
+			of vendors in each entry. Testing in production shows it works well enough.
+			**/
+			vendors = make(map[int]bool, 2*numEntries*(endVendorID-startOrOnlyVendorID+1))
```

Comment:
Why do you want to multiply it by 2? It should be enough to provide the capacity we need. 
```suggestion
			vendors = make(map[int]bool, numEntries*(endVendorID-startOrOnlyVendorID+1))
```